### PR TITLE
Michael and Brett's Changes

### PR DESCRIPTION
**Pull request by gmanjames**
I also added a LICENSE file and a README that we can begin editing. I suppose we can just add things to this pull request and merge it into master before midnight.

### CODE REVIEW

**Comment by gmanjames** [comment]
Be sure to resolve any merge conflicts before pushing up your changes I should note.

File: js/controls.js [hunk 1]
```diff
@@ -504,9 +545,9 @@ const Controls = () =>
         for (let i = 0; i < evt.target.parentNode.children.length; i++) {
             child = evt.target.parentNode.children[i];
             if (child === evt.target)
-            child.dataset.selected = true;
+                child.dataset.selected = true;
```
Comment (kylefinter) [inline js/controls.js:548]: Looks like still incorrect spacing.

File: js/controls.js [hunk 2]
```diff
@@ -921,6 +986,9 @@ An example path here is:\n\n :userName/:repoName/branchName/path/to/fileName.jso
         // Add event listener to toggle-floor
         winw.querySelector('.floor-toggle input').addEventListener('change', handleFloors);
 
+		// Add event listener to toggle-camera
+		winw.querySelector('.camera-toggle input').addEventListener('change', handleLockCamera);
```
Comment (kylefinter) [inline js/controls.js]: Also incorrect spacing here.

File: js/visualizer.js [hunk 3]
```diff
@@ -81,15 +97,35 @@ const Visualizer = (fps) =>
...
      */
     let grid;
 
     /*
-     * Default ground for the 3D environment
+     * Default ground for the 3D environment.
      */
     let ground;
 
+	/*
```
Comment (kylefinter) [inline js/visualizer.js]: Indented by too much here

File: js/visualizer.js [hunk 4]
```diff
@@ -107,55 +143,91 @@ const Visualizer = (fps) =>
...
         camera.position.set(600, 600, 1000);
         camera.lookAt(new THREE.Vector3(0,0,0));
 
-        // Directional light to enhance shadow.
-        let defaultLight = new THREE.DirectionalLight(0xffffff);
-        defaultLight.position.set(0, 1000, 0);
-        defaultLight.lookAt(new THREE.Vector3(0, 0, 0));
-        scene.add(defaultLight);
+		// Spotlight.
+		spotLight = new THREE.SpotLight();
```
Comment (kylefinter) [inline js/visualizer.js]: Incorrect Indent here and further down

File: js/visualizer.js [hunk 5]
```diff
@@ -107,55 +143,91 @@ const Visualizer = (fps) =>
...
-    		scene.fog = new THREE.Fog(0x001a0d, 1500, 4500);
-    		grid = new THREE.GridHelper(10000, 100, 0x001a0d, 0x006633);
-    		scene.add(grid);
+    	// Scene background.
+    	renderer.gammaInput = true;
+    	renderer.gammaOutput = true;
+    	renderer.setClearColor(0x001a0d, 1.0);
+    	renderer.shadowMap.enabled = true;
+		/*
+		let skyGeo = new THREE.SphereGeometry(3000, 60, 40);
```
Comment (kylefinter) [inline js/visualizer.js]: We shouldn't commit commented out code. Save this for later until we know we need it.

**Comment by gmanjames** [comment]
If you find anything go ahead and correct it. That way we have a better chance of hitting everything.

File: js/visualizer.js [hunk 6]
```diff
@@ -323,17 +397,26 @@ const Visualizer = (fps) =>
                 animation.frames[frame][group.name].quaternion[2],
                 animation.frames[frame][group.name].quaternion[3]
             );
+
+			//modelPos = group.position;
+			//lightTarg = group;
```
Comment (kylefinter) [inline js/visualizer.js]: And here incorrect indent

File: js/visualizer.js [hunk 7]
```diff
@@ -323,17 +397,26 @@ const Visualizer = (fps) =>
...
     /* ------------------------------------------------------------------------
      * render:
      * ------------------------------------------------------------------------
-     * ...
+     * Call Threejs function to render to the canvas.
      */
     function render()
     {
+		if (freeCam == false) {
+			camera.lookAt(modelPos);
```
Comment (kylefinter) [inline js/visualizer.js]: Marking for either me to fix or others

File: js/controls.js (hunk 1 above)
Comment (kylefinter) [reply js/controls.js:548]: Sorry this is correct.

**Review by kylefinter — APPROVED** [review]

**Comment by gmanjames** [comment]
I'm prepared to merge if everyone else is

**Pull request merged**